<b>Exercise 1</b> <i>A Circuit of Resistors</i>

1. Writing similar equations for all junction points
    
     $3V_1 - V_2 - V_3 - V_4 = 5$<br>
    $-V_1 + 4V_2 - V_3 - V_4 = 5$<br>
    $-V_1 - V_2 + 4V_3 - V_4 = 0$<br>
    $-V_1 - V_2 - V_3 + 3V_4 = 0$<br>
    
2. System of Linear equations in the form Av=b
    
        [[3 -1 -1 0 | 5]
         [-1 4 -1 -1 | 5]
         [-1 -1 4 -1 | 0]
         [0 -1 -1 3 | 0]]

In [3]:
#Kaleb Kautzsch

#Import scipy and numpy
import numpy as np
import scipy.linalg as la

#Create different matrices
A = np.matrix([[3, -1, -1, 0],[-1, 4, -1, -1],[-1, -1, 4, -1],[0, -1, -1, 3]])

b = np.matrix([[5],[5],[0],[0]])

#Use scipy to solve for the series of linear equations
v = la.solve(A, b)

print(v)

[[3.33333333]
 [3.        ]
 [2.        ]
 [1.66666667]]


<b>Exercise 2</b> <i>A chain of resistors</i>

In [9]:
#Kaleb Kautzsch

#Import programs
import numpy as np
import scipy.linalg as la

#Create matrices of zeros to solve for
N = 10 #Changeable variable

A = np.zeros([N,N])
b = np.zeros((N,1))

#Setting appropiate equal values to solve for equations
b[0,0] = 5
b[1,0] = 5

#A for loop to to run operations to solve for any [N,N] matrix
for i in range(N):
    for j in range (N):
        """This is the case when the equation subscripts are the same"""
        if i == j:
            A[i][j] = 4
        elif i == j-1: #Everything starting here down are for the different cases of N in the matrix
            A[i][j] = -1
        elif i == j+1:
            A[i][j] = -1
        elif i == j-2:
            A[i][j] = -1
        elif i == j+2:
            A[i][j] = -1

A[0,0] = 3
A[N-1,N-1] = 3
v = la.solve(A,b)

print(A)
print(v)

[[ 3. -1. -1.  0.  0.  0.  0.  0.  0.  0.]
 [-1.  4. -1. -1.  0.  0.  0.  0.  0.  0.]
 [-1. -1.  4. -1. -1.  0.  0.  0.  0.  0.]
 [ 0. -1. -1.  4. -1. -1.  0.  0.  0.  0.]
 [ 0.  0. -1. -1.  4. -1. -1.  0.  0.  0.]
 [ 0.  0.  0. -1. -1.  4. -1. -1.  0.  0.]
 [ 0.  0.  0.  0. -1. -1.  4. -1. -1.  0.]
 [ 0.  0.  0.  0.  0. -1. -1.  4. -1. -1.]
 [ 0.  0.  0.  0.  0.  0. -1. -1.  4. -1.]
 [ 0.  0.  0.  0.  0.  0.  0. -1. -1.  3.]]
[[4.12573674]
 [3.9194499 ]
 [3.45776031]
 [3.09430255]
 [2.69155206]
 [2.30844794]
 [1.90569745]
 [1.54223969]
 [1.0805501 ]
 [0.87426326]]


<b>Results for N = 4 and N = 10</b>
When N = 4:

                                    A = [[ 3. -1. -1.  0.]
                                         [-1.  4. -1. -1.]
                                         [-1. -1.  4. -1.]
                                         [ 0. -1. -1.  3.]]
                                         
                                    v = [[3.33333333]
                                         [3.        ]
                                         [2.        ]
                                         [1.66666667]]
                                         
This value for "v" is the same calculated in my first code.

When N = 10:

                                    A = [[ 3. -1. -1.  0.  0.  0.  0.  0.  0.  0.]
                                         [-1.  4. -1. -1.  0.  0.  0.  0.  0.  0.]
                                         [-1. -1.  4. -1. -1.  0.  0.  0.  0.  0.]
                                         [ 0. -1. -1.  4. -1. -1.  0.  0.  0.  0.]
                                         [ 0.  0. -1. -1.  4. -1. -1.  0.  0.  0.]
                                         [ 0.  0.  0. -1. -1.  4. -1. -1.  0.  0.]
                                         [ 0.  0.  0.  0. -1. -1.  4. -1. -1.  0.]
                                         [ 0.  0.  0.  0.  0. -1. -1.  4. -1. -1.]
                                         [ 0.  0.  0.  0.  0.  0. -1. -1.  4. -1.]
                                         [ 0.  0.  0.  0.  0.  0.  0. -1. -1.  3.]]
                                         
                                     v = [[4.12573674]
                                          [3.9194499 ]
                                          [3.45776031]
                                          [3.09430255]
                                          [2.69155206]
                                          [2.30844794]
                                          [1.90569745]
                                          [1.54223969]
                                          [1.0805501 ]
                                          [0.87426326]]